web/src/components/datasets/DatasetColumnLineage.tsx

```diff
@@ -0,0 +1,108 @@
+// SPDX-License-Identifier: Apache-2.0
+
+import React, { FunctionComponent, useEffect } from 'react'
+import * as Redux from 'redux'
+import { Box, Button } from '@material-ui/core'
+import MqEmpty from '../core/empty/MqEmpty'
+import MqJson from '../core/code/MqJson'
+import MqText from '../core/text/MqText'
+import { bindActionCreators } from 'redux'
+import { connect } from 'react-redux'
+import { Dataset } from '../../types/api'
+import { fetchDataset, resetDataset } from '../../store/actionCreators'
+import { fileSize } from '../../helpers'
+import { LineageDataset } from '../lineage/types'
+import { saveAs } from 'file-saver'
+import { IState } from '../../store/reducers'
+
+interface DatasetColumnLineageProps {
+  lineageDataset: LineageDataset
+}
+
+interface StateProps {
+  dataset: Dataset
+}
+
+interface DispatchProps {
+  fetchDataset: typeof fetchDataset
+  resetDataset: typeof resetDataset
+}
+
+type IProps = DatasetColumnLineageProps & DispatchProps & StateProps
+
+const DatasetColumnLineage: FunctionComponent<IProps> = props => {
+  const { dataset, lineageDataset, fetchDataset, resetDataset } = props
+  const columnLineage = dataset.columnLineage
+
+  useEffect(() => {
+    fetchDataset(lineageDataset.namespace, lineageDataset.name)
+  }, [lineageDataset.name])
+
+  // unmounting
+  useEffect(() => () => {
+    resetDataset()
+  }, [])
+
+  const handleDownloadPayload = (data: object) => {
+    const title = `${lineageDataset.name}-${lineageDataset.namespace}-columnLineage`
```

Comment:
We might want a file extension added here.

Reply:
Sure, but extension exist on 49 line (json). It's should be okay